src/Ombi.Core/Authentication/LdapUserManager.cs

```diff
@@ -0,0 +1,168 @@
+        {
+            try
+            {
+                return userEntry.GetAttribute(attr);
+            }
+            catch (Exception)
+            {
+                return null;
+            }
+        }
+
+        private async Task<LdapConnection> BindLdapConnection(string username, string password)
+        {
+            var settings = await GetSettings();
+
+            var ldapClient = new LdapConnection { SecureSocketLayer = settings.UseSsl };
+            try
+            {
+                if (settings.SkipSslVerify)
+                {
+                    ldapClient.UserDefinedServerCertValidationDelegate += LdapClient_UserDefinedServerCertValidationDelegate;
+                }
+
+                ldapClient.Connect(settings.Hostname, settings.Port);
+                if (settings.UseStartTls)
+                {
+                    ldapClient.StartTls();
+                }
+
+                ldapClient.Bind(username, password);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, "Failed to Connect or Bind to server");
+                throw e;
+            }
+            finally
+            {
+                ldapClient.UserDefinedServerCertValidationDelegate -= LdapClient_UserDefinedServerCertValidationDelegate;
+            }
+
+            if (!ldapClient.Bound)
+            {
+                ldapClient.Dispose();
+                return null;
+            }
+
+            return ldapClient;
+        }
+
+        private async Task<ILdapSearchResults> SearchLdapUsers(LdapConnection ldapClient)
+        {
+            var settings = await GetSettings();
+
+            string[] searchAttributes = { settings.UsernameAttribute };
+
+            _logger.LogDebug("Search: {1} {2} @ {3}", settings.BaseDn, settings.SearchFilter, settings.Hostname);
+            return ldapClient.Search(settings.BaseDn, LdapConnection.ScopeSub, settings.SearchFilter, searchAttributes, false);
+        }
+
+        public async Task<ILdapSearchResults> GetLdapUsers()
+        {
+            var settings = await GetSettings();
+
+            using var ldapClient = await BindLdapConnection(settings.BindUserDn, settings.BindUserPassword);
+
+            return await SearchLdapUsers(ldapClient);
+        }
+
+        public async Task<LdapEntry> LocateLdapUser(string username)
+        {
+            var settings = await GetSettings();
+
+            using var ldapClient = await BindLdapConnection(settings.BindUserDn, settings.BindUserPassword);
+            var ldapUsers = await SearchLdapUsers(ldapClient);
+            if (ldapUsers == null)
+            {
+                return null;
+            }
+
+            while (ldapUsers.HasMore())
+            {
+                var currentUser = ldapUsers.Next();
+                var foundUsername = GetLdapAttribute(currentUser, settings.UsernameAttribute)?.StringValue;
+                if (foundUsername == username)
```

Comment:
It would be better to do a `string.Equals` and use a case insensitive pattern match, see here: https://docs.microsoft.com/en-us/dotnet/api/system.string.equals?view=netcore-3.1#System_String_Equals_System_String_System_String_System_StringComparison_